### PR TITLE
docs: describe operational readiness

### LINKED ISSUE #12426: docs: Build public-facing documentation for Reference Architectures
Today, as part of our V2 Scale Tests, we are simulating a load much higher than current Coder usage, aiming for 2000 concurrent active users. This effort is crucial for identifying bugs and informing scaling recommendations in preparation for potential customer demand.

Considering our customers' aggressive scaling goals, including up to 5k known users, it's evident that our customers require assurance that Coder and their infrastructure can handle such user loads effectively. To support successful customer deployments, especially for thousands of users, it's essential to provide not only scalability but also other key ingredients.

Therefore, we need to develop public-facing documentation that outlines reference architectures and offers prescriptive solutions for different use cases. This documentation will not only benefit our customers but also streamline our onboarding process and provide a clearer roadmap for future scalability efforts.

_Architecture by Scale_ (phase 1):
- [x] https://github.com/coder/coder/pull/12438 Coder Glossary
  - admin, user, workspace, template, proxy, provisioner, registry, Kubernetes cluster for Coder
- [x] https://github.com/coder/coder/pull/12438 Describe scale tests methodology
  - infra/setup requirements, web terminal, applications
  - traffic projections during scale tests (continuous volume, req/s, users, aggressive approach)
- [x] https://github.com/coder/coder/pull/12534 Hardware recommendations for Coder control plane
  - node sizing, resource limits, number of replicas
    - refer to public cloud providers: AWS, GCP, Azure
  - reasonable ratio/formula: CPU x memory x users
  - reasonable ratio/formula: provisionerd x users
  - API latency/response time
  - average number of HTTP requests
  - advice: do not enable autoscaling
- [x] https://github.com/coder/coder/pull/12534 Hardware recommendations for Coder workspaces
  - Assumptions: 
    - workspaces also run on the same Kubernetes cluster (recommend a different namespace/node pool)
    - developers can pick between 4-8 CPU and 4-16 GB RAM workspaces (limits)
    - developers have a resource quota of 16 GPU 32 GB RAM (2-maxed out workspaces).
    - web microservice development use case: resources are mostly underutilized but spike during builds
   - Case study:
     - Developers for up to 2000+ users architecture are in 2 regions (a different cluster) and are evenly split. In practice, this doesn’t change much besides the diagram and workspaces node pool autoscaling config as it still uses the central provisioner. Recommend multiple provisioner groups for zero-trust and multi-cloud use cases.
    - Developers for up to 3000+ users architecture are also in an on-premises network. Document a provisioner running in a different cloud environment, and the zero-trust benefits of that.
- [x] https://github.com/coder/coder/pull/12534 Hardware sizing recommendations for PostgreSQL database
  - measure and document the impact of `dbcrypt`
- [x] https://github.com/coder/coder/pull/12643 Describe how to run scale tests against our own infrastructure
  - use coder CLI, link dedicated command docs
  - expose to public Coder internal template for scale tests
  - expose to public Grafana dashboards
- [ ] https://github.com/coder/coder/pull/12723 Operational Readiness
  - Configuration: Helm values, environment variables
  - Template configuration managed via CI/CD.
  - Observability, e.g. Promethus/Grafana
  - Database backups
    - link scripts if we have any
  - Custom support links
  - Custom agent troubleshooting links
- [ ] Network recommendations for Coder
  - control plane ↔ workspaces ↔ clients

TODO:
- [x] Mention that as users onboard, the autoscaling config should take care of ongoing workspaces

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/12426

This PR extends Reference Architectures with a paragraph about the operational readiness of Coder deployment. Feel free to add more ideas to the checklist.

Extra changes:
* do no use parallel mode while running `make gen` and `make build/coder_docs...`. It fails with `coderd/rbac/object_gen.go:1:1: syntax error: package statement must be first`.

### CODE REVIEW

File: docs/admin/architectures/index.md [hunk 1]
```diff
@@ -342,3 +342,52 @@ We provide the following general recommendations for PostgreSQL settings:
...
+needs. Operational readiness is crucial because it helps prevent issues that
+could affect workspace users experience once the platform is live.
+
+Learn about Coder design principles and architectural best practices described
+in the
+[Well-Architected Framework](https://coder.com/blog/coder-well-architected-framework).
+
+### Configuration
+
+1. Create `values.yaml` and add it to a version control system.
```
Comment (johnstcn) [inline docs/admin/architectures/index.md:360]:
```suggestion
1. Create `values.yaml` and add it to a version control system.
   _Note:_ it is highly recommended that you create a custom `values.yaml` as opposed to copying the entire default values.
```

File: docs/admin/architectures/index.md [hunk 2]
```diff
@@ -342,3 +342,52 @@ We provide the following general recommendations for PostgreSQL settings:
...
+1. Determine the necessary environment variables.
+1. Identify the required Helm values for configuration.
+
+### Template configuration
+
+1. Establish a dedicated user account for the _Template Administrator_.
+1. Maintain Coder templates using version control.
+1. Consider implementing a GitOps workflow to automatically push new template
```
Comment (johnstcn) [inline docs/admin/architectures/index.md:368]:
```suggestion
1. Consider implementing a GitOps workflow to automatically push new template.
   For example, on Github, you can use the [Update Coder Template](https://github.com/marketplace/actions/update-coder-template) action.
```

File: docs/admin/architectures/index.md [hunk 3]
```diff
@@ -342,3 +342,52 @@ We provide the following general recommendations for PostgreSQL settings:
...
+
+1. Prepare internal scripts for dumping and restoring databases.
+1. Schedule regular database backups, especially before release upgrades.
+
+### User support
+
+1. Incorporate [support links](../appearance.md#support-links) into internal
+   documentation accessible from the user context menu. Ensure that hyperlinks
+   are valid and lead to up-to-date materials.
+1. Encourage the use of `coder support bundle` to allow workspace users to
```
Comment (johnstcn) [inline docs/admin/architectures/index.md:392]: `coder support bundle` can only be used by administrators -- additionally, the command is currently hidden and will be un-hidden closer to the next release.

File: docs/admin/architectures/index.md [hunk 4]
```diff
@@ -342,3 +342,52 @@ We provide the following general recommendations for PostgreSQL settings:
...
+
+Learn about Coder design principles and architectural best practices described
+in the
+[Well-Architected Framework](https://coder.com/blog/coder-well-architected-framework).
+
+### Configuration
+
+1. Create `values.yaml` and add it to a version control system.
+1. Determine the necessary environment variables.
+1. Identify the required Helm values for configuration.
```
Comment (johnstcn) [inline docs/admin/architectures/index.md:362]: nit: this might come before creating a values.yaml

File: docs/admin/architectures/index.md [hunk 5]
```diff
@@ -342,3 +342,52 @@ We provide the following general recommendations for PostgreSQL settings:
...
+
```
Comment (johnstcn) [inline docs/admin/architectures/index.md:363]:
```suggestion

### Deployment

1. Leverage automation tooling to automate deployment and upgrades of Coder.

```

File: docs/admin/architectures/index.md (hunk 3 above)
Comment (mtojek) [reply docs/admin/architectures/index.md:400]: I guess that this is coming now 👍

**Review by johnstcn — APPROVED** [review]

**Pull request merged**